File: services/analytics-service-flask/app/services/technology_trace_service.py
class TechnologyTraceService:

    @staticmethod
    def get_trace(action, request_path):
        return {
            "action": action,
            "service": "Healthcare Analytics Service",
            "module": "analytics-service-flask",
            "technologyStack": ["Flask"],
            "businessDomain": "Healthcare Analytics",
            "whyThisStack": "Flask is used for lightweight analytics and KPI reporting endpoints due to its fast execution and micro-footprint.",
            "requestPath": request_path,
            "architectureRole": "Exposes performance indexes and dashboard reporting metrics."
        }

File: services/risk-scoring-service-scala/src/main/scala/services/TechnologyTraceService.scala
package com.example.polymed.riskservice.services

import com.example.polymed.riskservice.models.TechnologyTrace

object TechnologyTraceService {
  def getTrace(requestPath: String): TechnologyTrace = {
    TechnologyTrace(
      action = "EVALUATE_RISK",
      service = "Clinical Risk Scoring Service",
      module = "risk-scoring-service-scala",
      technologyStack = List("Scala"),
      businessDomain = "Clinical Risk Scoring",
      whyThisStack = "Scala is used for clinical risk scoring because of its pure functional rule evaluation capabilities, type-safety, and side-effect-free math execution.",
      requestPath = requestPath,
      architectureRole = "Domain service responsible for executing clinical algorithms on patient metrics."
    )
  }
}
